<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot Differentially Methylated Regions — plotDMRs2 • DMRichR</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Plot Differentially Methylated Regions — plotDMRs2" />
<meta property="og:description" content="Generates trace plots of methylation proportions by genomic position. Modified
from the plotDMRs function in the dmrseq R package (plotDMRs)." />
<meta property="og:image" content="https://ben-laufer.github.io/DMRichR/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DMRichR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/DMRichR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ben-laufer/DMRichR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot Differentially Methylated Regions</h1>
    <small class="dont-index">Source: <a href='https://github.com/ben-laufer/DMRichR/blob/master/R/plotDMRs2.R'><code>R/plotDMRs2.R</code></a></small>
    <div class="hidden name"><code>plotDMRs2.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generates trace plots of methylation proportions by genomic position. Modified
from the plotDMRs function in the dmrseq R package (<code><a href='https://rdrr.io/pkg/dmrseq/man/plotDMRs.html'>plotDMRs</a></code>).</p>
    </div>

    <pre class="usage"><span class='fu'>plotDMRs2</span><span class='op'>(</span>
  <span class='va'>BSseq</span>,
  regions <span class='op'>=</span> <span class='cn'>NULL</span>,
  testCovariate <span class='op'>=</span> <span class='cn'>NULL</span>,
  extend <span class='op'>=</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/start.html'>end</a></span><span class='op'>(</span><span class='va'>regions</span><span class='op'>)</span> <span class='op'>-</span> <span class='fu'><a href='https://rdrr.io/r/stats/start.html'>start</a></span><span class='op'>(</span><span class='va'>regions</span><span class='op'>)</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span><span class='op'>/</span><span class='fl'>2</span>,
  main <span class='op'>=</span> <span class='st'>""</span>,
  addRegions <span class='op'>=</span> <span class='va'>regions</span>,
  annoTrack <span class='op'>=</span> <span class='cn'>NULL</span>,
  col <span class='op'>=</span> <span class='cn'>NULL</span>,
  lty <span class='op'>=</span> <span class='cn'>NULL</span>,
  lwd <span class='op'>=</span> <span class='cn'>NULL</span>,
  label <span class='op'>=</span> <span class='cn'>NULL</span>,
  mainWithWidth <span class='op'>=</span> <span class='cn'>TRUE</span>,
  regionCol <span class='op'>=</span> <span class='st'>"#FF00001A"</span>,
  addTicks <span class='op'>=</span> <span class='cn'>TRUE</span>,
  addPoints <span class='op'>=</span> <span class='cn'>TRUE</span>,
  pointsMinCov <span class='op'>=</span> <span class='fl'>1</span>,
  highlightMain <span class='op'>=</span> <span class='cn'>FALSE</span>,
  qval <span class='op'>=</span> <span class='cn'>TRUE</span>,
  stat <span class='op'>=</span> <span class='cn'>TRUE</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>,
  includeYlab <span class='op'>=</span> <span class='cn'>TRUE</span>,
  compareTrack <span class='op'>=</span> <span class='cn'>NULL</span>,
  labelCols <span class='op'>=</span> <span class='cn'>NULL</span>,
  horizLegend <span class='op'>=</span> <span class='cn'>FALSE</span>,
  addLines <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>BSseq</th>
      <td><p>An object of class BSseq. Can be smoothed or not smoothed.</p></td>
    </tr>
    <tr>
      <th>regions</th>
      <td><p>A data.frame containing the DMRs (output from the main
<code>dmrseq</code> function).</p></td>
    </tr>
    <tr>
      <th>testCovariate</th>
      <td><p>integer value or vector indicating which of columns of
<code>pData(bs)</code> contains the covariate of interest. 
This is used to construct the sample labels and colors (unless this is
over-ridden by specifying <code>label</code>).</p></td>
    </tr>
    <tr>
      <th>extend</th>
      <td><p>Describes how much the plotting region should be extended in
either direction. The total width of the plot is equal to the width of the 
region plus twice extend.</p></td>
    </tr>
    <tr>
      <th>main</th>
      <td><p>The plot title. The default is to construct a title with 
information about which genomic region is being plotted.</p></td>
    </tr>
    <tr>
      <th>addRegions</th>
      <td><p>A set of additional regions to be highlighted on the 
plots. Same format as the <code>regions</code> argument.</p></td>
    </tr>
    <tr>
      <th>annoTrack</th>
      <td><p>a <code>GRangesList</code> object with two elements returned
by <code>getAnnot</code>. The first
contains CpG category information in the first element (optional)
coding gene sequence information in the second element (optional).
At least one of these elements needs to be non-null in order for 
any annotation to be plotted, but it is not necessary to contain
both.</p></td>
    </tr>
    <tr>
      <th>col</th>
      <td><p>The color of the methylation estimates. It is recommended to 
leave this value as default (NULL), and specify a value of 
<code>testCovariate</code> to indicate which column of <code>pData(bs)</code>
to use as a factor for coloring the points and lines of the plot.
Alternatively, you can specify particular colors by 
passing this information through the <code>pData</code> slot of the
object <code>BSseq</code> (a <code>data.frame</code> that houses metadata). To do 
so, place the color value for each sample in a column titled <code>col</code>, 
and leave this argument as its default value of NULL. Alternatively,
you may specify a vector of color values (one for each sample), but 
you *must* make sure that this vector is in the same order as the samples
are in the BSseq object. If NULL and no <code>col</code> column is found in 
<code>pData</code>, then estimates are plotted in black for all samples.</p></td>
    </tr>
    <tr>
      <th>lty</th>
      <td><p>The line type of the methylation estimates. It is recommended to
pass this information through the <code>pData</code> slot of the
object <code>BSseq</code> (a <code>data.frame</code> that houses metadata). To do 
so, place the line type value for each sample in a column titled <code>lty</code>, 
and leave this argument as its default value of NULL. Alternatively,
you may specify a vector of line type values (one for each sample), but 
you *must* make sure that this vector is in the same order as the samples
are in the BSseq object. If NULL and no <code>lty</code> column is found in 
<code>pData</code>, then estimates are plotted with <code>lty=1</code> for all samples.</p></td>
    </tr>
    <tr>
      <th>lwd</th>
      <td><p>The line width of the methylation estimates. It is recommended to
pass this information through the <code>pData</code> slot of the
object <code>BSseq</code> (a <code>data.frame</code> that houses metadata). To do 
so, place the line width value for each sample in a column titled <code>lwd</code>,
and leave this argument as its default value of NULL. Alternatively,
you may specify a vector of line width values (one for each sample), but 
you *must* make sure that this vector is in the same order as the samples
are in the BSseq object. If NULL and no <code>lwd</code> column is found in 
<code>pData</code>, then estimates are plotted with <code>lwd=1</code> for all samples.</p></td>
    </tr>
    <tr>
      <th>label</th>
      <td><p>The condition/population labels for the plot legend. If NULL
(default) this is taken from the <code>testCovariate</code> column of 
<code>pData</code>. Alternatively, you can pass in labels by 
adding this information through the <code>pData</code> slot of the
object <code>BSseq</code> (a <code>data.frame</code> that houses metadata). To do 
so, place the labels for each sample in a column titled <code>label</code>, 
and leave this argument as its default value of NULL.
You may instead specify an arbitrary vector of labels (one for each sample),
but be aware that you *must* make sure that this vector is in the same order 
as the samples are in the BSseq object. If NULL, and <code>testCovariate</code> is
also NULL and no <code>label</code> column is found in 
<code>pData</code>, then no legend is created.</p></td>
    </tr>
    <tr>
      <th>mainWithWidth</th>
      <td><p>logical value indicating whether the default title
should include information about width of the plot region.</p></td>
    </tr>
    <tr>
      <th>regionCol</th>
      <td><p>The color used for highlighting the region.</p></td>
    </tr>
    <tr>
      <th>addTicks</th>
      <td><p>logical value indicating whether tick marks showing the
location of methylation loci should be added. Default is TRUE.</p></td>
    </tr>
    <tr>
      <th>addPoints</th>
      <td><p>logical value indicating whether the individual 
methylation estimates be plotted as points.</p></td>
    </tr>
    <tr>
      <th>pointsMinCov</th>
      <td><p>The minimum coverage a methylation loci need in 
order for the raw methylation estimates to be plotted. Useful for filtering
out low coverage loci. Only used if addPoints = TRUE. Default value is 1 
(no filtering).</p></td>
    </tr>
    <tr>
      <th>highlightMain</th>
      <td><p>logical value indicating whether the plot region 
should be highlighted.</p></td>
    </tr>
    <tr>
      <th>qval</th>
      <td><p>logical value indicating whether the region FDR estimate
(q-value) should be displayed in the plot title. The value is extracted 
from the <code>regions</code> argument.</p></td>
    </tr>
    <tr>
      <th>stat</th>
      <td><p>logical value indicating whether the region statistic 
should be displayed in the plot title. The value is extracted from the
<code>regions</code> argument.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical value indicating whether progress messages 
should be printed to the screen.</p></td>
    </tr>
    <tr>
      <th>includeYlab</th>
      <td><p>a logical indicating whether to include the Y axis
label 'Methylation' (useful to turn off if combining multiple region
figures and you do not want to include redundant y axis label information)</p></td>
    </tr>
    <tr>
      <th>compareTrack</th>
      <td><p>a named GenomicRangesList item that contains up to four
custom tracks (GenomicRanges objects) which will be plotted below the region.
Only one of `compareTrack` or `annoTrack` can be specified since there is
only for plotting either the built in GpG category and exon tracks, *or* a
custom set of tracks.</p></td>
    </tr>
    <tr>
      <th>labelCols</th>
      <td><p>a character vector with names of the mcols slot of the 
GenomicRanges items in `compareTrack'. Only used if plotting custom
tracks using the `compareTrack' argument. If specified, the (first) value
in that column is printed along with a label that includes the name of the
list item. If NULL (default), just the name of the track is printed.</p></td>
    </tr>
    <tr>
      <th>horizLegend</th>
      <td><p>logical indicating whether the legend should be 
horizontal instead of vertical (default FALSE). This is useful if you need
to plot many labels and want to preserve whitespace.</p></td>
    </tr>
    <tr>
      <th>addLines</th>
      <td><p>logical indicating whether to plot smooth lines between 
points. Default is true. Can be useful to turn this off for very small 
regions.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>None (generates a plot)</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Creates aesthetially pleasing DMR plots. By default will plot individual
points with size proportional to coverage, along with a smoothed line 
for each sample. If BSseq object is smoothed, it will use these values. 
Otherwise, raw methylation values will be smoothed with loess. Elements 
will be colored by biological condition (<code>label</code>). Also has 
functionality to add annotations below the main plot (CpG category, genes) 
if <code>annoTrack</code> is specified.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p><code><a href='https://rdrr.io/pkg/dmrseq/man/plotDMRs.html'>plotDMRs</a></code></p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Ben Laufer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


